### PR TITLE
Update samples to Akka 2.6 and Typed - sharding #132

### PR DESCRIPTION
**Pull request by helena**
* #132 - sharding sample
* Squashed for a clean diff https://github.com/akka/akka-samples/pull/136 - which was meant to be a draft first

### CODE REVIEW

File: akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  def apply(system: ActorSystem[_]): FogSettings = {
+    apply(system.settings.config.getConfig("killrweather"))
+  }
+
+  def apply(config: Config): FogSettings = {
+    import akka.util.Helpers.Requiring
+
+    val config =
+      ConfigFactory.load("application.conf").getConfig("killrweather")
```
Comment (patriknw) [inline akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala]: strange to load the config again, when it's already passed in as parameter

**Comment by helena** [comment]
@johanandren Not really, there's a nice to have curl to get aggregates but the sample is to show sharding. I think adding the curl-ability is fine to add after.

File: akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala (hunk 1 above)
Comment (helena) [reply akka-sample-sharding-scala/killrweather-fog/src/main/scala/sample/killrweather/fog/Fog.scala]: Oops that's a line I forgot to finish refactoring, fixing.

File: akka-sample-sharding-scala/build.sbt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-val AkkaVersion = "2.6.0-RC2"
+val AkkaVersion = "2.6.0"
```
Comment (helena) [inline akka-sample-sharding-scala/build.sbt]: w00t!

**Comment by helena** [comment]
Merging with config line removed https://github.com/akka/akka-samples/pull/148/commits/366aa3ea456a3a8f12b5cb2b2c94855a981e032a#diff-9a2d542dc329db0469b69eb2ed41f6d2L56

**Pull request merged**